### PR TITLE
i1480: Tighten up data types on burden_estimate tables 

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-1480

Need to merge https://github.com/vimc/montagu-db/pull/69 first and then return `db_version` to master.

### CODE REVIEW

File: src/databaseInterface/src/main/java/org/vaccineimpact/api/db/Sequences.java [hunk 1]
```diff
@@ -28,11 +28,6 @@
      */
     public static final Sequence<Long> API_ACCESS_LOG_ID_SEQ = new SequenceImpl<Long>("api_access_log_id_seq", Public.PUBLIC, org.jooq.impl.SQLDataType.BIGINT.nullable(false));
 
-    /**
-     * The sequence <code>public.burden_estimate_id_seq</code>
-     */
-    public static final Sequence<Long> BURDEN_ESTIMATE_ID_SEQ = new SequenceImpl<Long>("burden_estimate_id_seq", Public.PUBLIC, org.jooq.impl.SQLDataType.BIGINT.nullable(false));
-
```
Comment (MartinEden) [inline src/databaseInterface/src/main/java/org/vaccineimpact/api/db/Sequences.java]: This shouldn't have vanished - suggests primary key is no longer auto-incrementing

File: src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/Indexes.java [hunk 2]
```diff
@@ -40,6 +40,6 @@
...
-        public static Index BURDEN_ESTIMATE_STOCHASTIC_UNIQUE = Internal.createIndex("burden_estimate_stochastic_unique", BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC, new OrderField[] { BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_ESTIMATE_SET, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.MODEL_RUN, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.COUNTRY, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.YEAR, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.AGE, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_OUTCOME }, true);
+        public static Index BURDEN_ESTIMATE_STOCHASTIC_UNIQUE = Internal.createIndex("burden_estimate_stochastic_unique", BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC, new OrderField[] { BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_ESTIMATE_SET, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.COUNTRY, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.YEAR, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.AGE, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_OUTCOME }, true);
```
Comment (MartinEden) [inline src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/Indexes.java]: We should think about the ordering on the unique constraint - although JOOQ may be getting it wrong, so the migration is the thing to examine

File: src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/Keys.java [hunk 3]
```diff
@@ -55,6 +55,6 @@
...
-        public static final UniqueKey<BurdenEstimateStochasticRecord> BURDEN_ESTIMATE_STOCHASTIC_UNIQUE = Internal.createUniqueKey(BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC, "burden_estimate_stochastic_unique", BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_ESTIMATE_SET, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.MODEL_RUN, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.COUNTRY, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.YEAR, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.AGE, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_OUTCOME);
+        public static final UniqueKey<BurdenEstimateStochasticRecord> BURDEN_ESTIMATE_STOCHASTIC_UNIQUE = Internal.createUniqueKey(BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC, "burden_estimate_stochastic_unique", BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_ESTIMATE_SET, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.COUNTRY, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.YEAR, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.AGE, BurdenEstimateStochastic.BURDEN_ESTIMATE_STOCHASTIC.BURDEN_OUTCOME);
```
Comment (MartinEden) [inline src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/Keys.java]: We should think about the ordering on the unique constraint - although JOOQ may be getting it wrong, so the migration is the thing to examine

File: src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/tables/BurdenEstimateStochastic.java [hunk 4]
```diff
@@ -87,12 +87,7 @@
     /**
      * The column <code>public.burden_estimate_stochastic.age</code>.
      */
-    public final TableField<BurdenEstimateStochasticRecord, Integer> AGE = createField("age", org.jooq.impl.SQLDataType.INTEGER, this, "");
-
-    /**
-     * The column <code>public.burden_estimate_stochastic.model_run</code>.
-     */
-    public final TableField<BurdenEstimateStochasticRecord, Integer> MODEL_RUN = createField("model_run", org.jooq.impl.SQLDataType.INTEGER.nullable(false), this, "");
+    public final TableField<BurdenEstimateStochasticRecord, Short> AGE = createField("age", org.jooq.impl.SQLDataType.SMALLINT, this, "");
```
Comment (MartinEden) [inline src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/tables/BurdenEstimateStochastic.java]: Why has this vanished?

Comment (MartinEden) [reply src/databaseInterface/src/main/java/org/vaccineimpact/api/db/annex/tables/BurdenEstimateStochastic.java]: Fixed

File: src/databaseTests/src/test/kotlin/org/vaccineimpact/api/databaseTests/tests/TriggerTests.kt [hunk 5]
```diff
@@ -45,7 +45,7 @@ class TriggerTests : DatabaseTest()
         return db.dsl.newRecord(table).apply {
             id = 1
             burdenEstimateSet = 0
-            country = "???"
+            country = 1
```
Comment (littledice) [inline src/databaseTests/src/test/kotlin/org/vaccineimpact/api/databaseTests/tests/TriggerTests.kt]: make this an invalid value to check triggers not enabled

**Comment by MartinEden** [comment]
Need to merge [the webmodels PR](https://github.com/vimc/montagu-webmodels/pull/45) first and then return API to pointing at submodule master.

**Pull request merged**